src/errorHandler/errorHandler.ts

```diff
@@ -1,68 +1,143 @@
-import { GraphQLError } from 'graphql';
-import { ApolloError } from 'apollo-server-errors';
+import {
+  GraphQLError,
+  GraphQLFormattedError,
+  GraphQLErrorOptions,
+} from 'graphql';
+import { unwrapResolverError } from '@apollo/server/errors';
+
+/**
+ * Internally managed error codes.  If a new error is added here,
+ * it should be added to `NO_REPORT_ERRORS` in `sentryPlugin` if
+ * we do not want to report it to sentry.
+ */
+export enum InternalErrorCode {
+  BAD_USER_INPUT = 'BAD_USER_INPUT',
+  FORBIDDEN = 'FORBIDDEN',
+  INTERNAL_SERVER_ERROR = 'INTERNAL_SERVER_ERROR',
+  NOT_FOUND = 'NOT_FOUND',
+  UNAUTHENTICATED = 'UNAUTHENTICATED',
+}
 
 /**
  * Used for formatting errors returned to the client. Hide any
  * errors that might reveal server details. Handle special cases
  * that we want to use to provide more information to the client
  * (e.g. NotFoundError).
  */
-export function errorHandler(error: GraphQLError): GraphQLError {
-  if (error.originalError instanceof NotFoundError) {
-    return new GraphQLNotFoundError(error);
-  } else if (
-    error instanceof ApolloError ||
-    error.originalError instanceof ApolloError
-  ) {
+export function errorHandler(
+  formattedError: GraphQLFormattedError,
+  error: unknown
+): GraphQLFormattedError {
+  if (unwrapResolverError(error) instanceof GraphQLError) {
     // Keep GraphQL errors intact
     // e.g. failed parsing, bad input
-    return error;
+    return formattedError;
   } else {
     // Mask other kinds of errors
-    return new InternalServerError(error);
+    return new InternalServerError('Internal server error', {
+      originalError: error as Error,
+      // preserve path and locations
+      path: (error as GraphQLError)?.path,
+      positions: (error as GraphQLError)?.positions,
+      source: (error as GraphQLError)?.source,
+    }).toJSON();
   }
 }
 
-export class NotFoundError extends Error {
-  static errorPrefix = `Error - Not Found`;
-  constructor(message?: string) {
-    super(`${NotFoundError.errorPrefix}: ${message}`); // 'Error' breaks prototype chain here
-    Object.setPrototypeOf(this, new.target.prototype); // restore prototype chain
+/**
+ * CustomGraphQLError exists for providing common extensions for all
+ * custom internal errors. Otherwise, all implementation is just relying
+ * on GraphQLError
+ *
+ * This really shouldn't be used directly.
+ */
+export class CustomGraphQLError extends GraphQLError implements Error {
```

Comment:
old class...
we can remove `export` and add comment on its deprecated

Reply:
I think there are reasons to still export. Updated this comment with more details.